### PR TITLE
Improvements to align CTS and Spec for Memory

### PR DESCRIPTION
**Pull request by martygrant**
- Add test for expecting UR_RESULT_ERROR_INVALID_NULL/HOST_POINTER for urMemImageCreate and urMemBufferCreate
- Add more checks for UR_RESULT_ERROR_INVALID_IMAGE_FORMAT_DESCRIPTOR for urMemImageCreate
- Added missing error condition to spec for urMemImageCreate - UR_STRUCTURE_TYPE_IMAGE_DESC != pImageDesc->stype
- Add tests for expecting UR_RESULT_ERROR_INVALID_NULL_HANDLE/POINTER for urMemBuffer/ImageCreateWithNativeHandle
- Remove skip for urMemImageCreateWithNativeHandle - buffer was invalid, creating it in the test fixture definition now
- Update image format used in urMemImageText fixture as it was invalid and would cause previously skipped tests to fail (they now pass)
- Fix DDI table entry for urMemImageCreateWithNativeHandle for OpenCL
- Add test for calling urMemBufferPartition with different ur_mem_flag_t flags
- Add missing UR_MEM_INFO_REFERENCE_COUNT to spec for urMemGetInfo. Added a test for this and checking ref count is updated in urMemRetain/Release

### CODE REVIEW

File: test/conformance/testing/include/uur/fixtures.h [hunk 1]
```diff
@@ -226,11 +230,12 @@ struct urMemImageTest : urContextTest {
         /*.height =*/16,
         /*.depth =*/1,
         /*.arraySize =*/1,
-        /*.rowPitch =*/16 * sizeof(char[4]),
-        /*.slicePitch =*/16 * 16 * sizeof(char[4]),
+        /*.rowPitch =*/0,   //16 * sizeof(char[4]),
```
Comment (RossBrunton) [inline test/conformance/testing/include/uur/fixtures.h:233]: Why is this commented out?

Comment (martygrant) [reply test/conformance/testing/include/uur/fixtures.h:233]: Nice spot, yeah that's a mistake during debugging. Slipped my todo list when cleaning up this branch. Will also remove the commented out test in urMemBufferCreateWithNativeHandle.cpp

Comment (martygrant) [reply test/conformance/testing/include/uur/fixtures.h]: Updated this, according [to the spec](https://oneapi-src.github.io/unified-runtime/core/api.html#urmemimagecreate) rowPitch and slicePitch should be 0

**Review by RossBrunton — APPROVED** [review]

File: scripts/core/memory.yml [hunk 2]
```diff
@@ -503,7 +509,7 @@ returns:
 type: function
 desc: "Create runtime image memory object from native memory handle."
 class: $xMem
-name: ImageCreateWithNativeHandle
+name: Image`ativeHandle
```
Comment (hdelan) [inline scripts/core/memory.yml:512]: Typo?

**Review by hdelan — APPROVED** [review]
Small typo. Otherwise LGTM

File: scripts/core/memory.yml (hunk 2 above)
Comment (martygrant) [reply scripts/core/memory.yml]: Thanks Hugh! Fixed this now.

**Review by PietroGhg — APPROVED** [review]
Native CPU lgtm, thank you

**Pull request closed without merging**